File: database/test-connection.py
#!/usr/bin/env python3
# Test database connection
# Run with: python database/test-connection.py

from db import test_connection, query

def test():
    print('Testing database connection...\n')
    
    if test_connection():
        print('\n✅ Connection successful! Testing a query...\n')
        
        try:
            # Test query - get all table names
            tables = query('SHOW TABLES')
            print('📊 Tables in database:')
            for index, table in enumerate(tables, 1):
                table_name = list(table.values())[0]
                print(f'   {index}. {table_name}')
            
            print(f'\n✅ Found {len(tables)} tables. Database is ready to use!')
        except Exception as error:
            print(f'❌ Query test failed: {error}')

if __name__ == '__main__':
    test()

File: database/test-connection.js
// Test database connection
// Run with: node database/test-connection.js

const db = require('./db');

async function test() {
  console.log('Testing database connection...\n');
  
  const connected = await db.testConnection();
  
  if (connected) {
    console.log('\n✅ Connection successful! Testing a query...\n');
    
    try {
      // Test query - get all table names
      const tables = await db.query('SHOW TABLES');
      console.log('📊 Tables in database:');
      const tableNames = tables.map(row => Object.values(row)[0]);
      tableNames.forEach((name, index) => {
        console.log(`   ${index + 1}. ${name}`);
      });
      
      console.log(`\n✅ Found ${tableNames.length} tables. Database is ready to use!`);
    } catch (error) {
      console.error('❌ Query test failed:', error.message);
    }
  }
  
  // Close pool
  process.exit(connected ? 0 : 1);
}

test();

